controllers/operatorconfigmap_controller.go

```diff
@@ -91,19 +93,19 @@ func (c *ClusterVersionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		),
 	)
 	// Reconcile the ClusterVersion object when the operator config map is updated
-	enqueueClusterVersionRequest := handler.EnqueueRequestsFromMapFunc(
+	enqueueConfigMapRequest := handler.EnqueueRequestsFromMapFunc(
 		func(_ context.Context, client client.Object) []reconcile.Request {
 			return []reconcile.Request{{
 				NamespacedName: types.NamespacedName{
-					Name: clusterVersionName,
+					Name: operatorConfigMapName,
 				},
 			}}
 		},
 	)
 
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&configv1.ClusterVersion{}, clusterVersionPredicates).
-		Watches(&corev1.ConfigMap{}, enqueueClusterVersionRequest, configMapPredicates).
+		For(&corev1.ConfigMap{}, configMapPredicates).
+		Watches(&configv1.ClusterVersion{}, enqueueConfigMapRequest, clusterVersionPredicates).
```

Comment:
not entirely sure if we still need to watch the ClusterVersion resource, was added to be on the safe side.

Reply:
JFYI: ClusterVersion watcher was added to ensure we update the csi sidecar images as per the ocp cluster version where its running.

Reply:
@bernerhat Yes we do, we need to peek changes to the cluster version as an additional input and to issue reconciles to make sure we are still running to correct version of the csi sidecar, as @Madhu-1 mentioned in his comment

Reply:
ack